pkg/cloud/openstack/services/networking/floatingip.go

```diff
@@ -0,0 +1,63 @@
+package networking
+
+import (
+	"fmt"
+	"github.com/gophercloud/gophercloud"
+	"github.com/gophercloud/gophercloud/openstack/networking/v2/extensions/layer3/floatingips"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/klog"
+	providerv1 "sigs.k8s.io/cluster-api-provider-openstack/pkg/apis/openstackproviderconfig/v1alpha1"
+	"time"
+)
+
+func (s *Service) CreateFloatingIP(clusterProviderSpec *providerv1.OpenstackClusterProviderSpec, ip string) error {
```

Comment:
suggest we rename this to GetorCreateFloatingIP as this might just return if it exist

Reply:
Yup